VDF.Core/ScanEngine.cs

```diff
@@ -211,7 +250,15 @@ public sealed class ScanEngine {
 
 		public static Task<bool> LoadDatabase() => Task.Run(DatabaseUtils.LoadDatabase);
 		public static void SaveDatabase() => DatabaseUtils.SaveDatabase();
-
+		public static void RemoveFromDatabase(FileEntry dbEntry) => DatabaseUtils.Database.Remove(dbEntry);
+		public static void UpdateFilePathInDatabase(String newPath, FileEntry dbEntry) {
+			DatabaseUtils.Database.Remove(dbEntry);
+			dbEntry.Path = newPath;
+			DatabaseUtils.Database.Add(dbEntry);
+		}
+		#pragma warning disable CS8601 // Possible null reference assignment
+		public static Boolean GetFromDatabase(String path, out FileEntry dbEntry) => DatabaseUtils.Database.TryGetValue(new FileEntry(path), out dbEntry);
+		#pragma warning restore CS8601 // Possible null reference assignment
```

Comment:
This is only used for updating file path. I'm not feeling well with a method which first gets the item. Then calling another method which removes this item, setting a new file path to this item only to re-add this item. Why don't we just have one method calling to update the file path? And this one just updates the item without removing it?

Reply:
Path is major part of `FileEntry` hashcode, i was pretty sure that to keep everything consistent it is required to re-add object with changed hashcode to the `HashSet` which is here implementation of the Database, but i will double check it then.
Also probably i should move this function to DatabaseUtils, not sure why I have not done this in the first place.

Regarding types i will adjust :) thanks for pointing this out.

Reply:
Ok, so i've just squashed all commits in these PR, and adjusted `String` and `Boolean` as requested.
I have also moved UpdateFilePathInDatabase body to DatabaseUtils.
Though regarding logic of this function from what I have checked it must stays that way, or whole Database needs to be redesigned, here is simplified example that should visualize what's the issue:

```
using System;
using System.Collections.Generic;

public class Program
{
	public static void Main()
	{
		FileEntry a1 = new FileEntry("a1");
		FileEntry a2 = new FileEntry("a2");
		HashSet<FileEntry> Database = new HashSet<FileEntry>();
		Database.Add(a1);
		Database.Add(a2);
		Database.Add(a1);

		Console.WriteLine(Database.Count); // Expected output: "2" because we are trying to add the same item twice

		a1.Path = "b1"; // let's do regular update that should work with for example list

		Database.TryGetValue(new FileEntry("b1"), out FileEntry b1);
		Console.WriteLine($"b1='{b1}'");  //Expected output: "_b1=''"  because hashcode of b1 is suggesting to use hash bucket in which hashset can't find adjusted a1 object

		Database.Add(a1);
		Console.WriteLine(Database.Count); // Expected output "3" because hashcode of a1 has changed

		Database.TryGetValue(new FileEntry("b1"), out b1);
		Console.WriteLine($"b1='{b1}'");  //Expected output: _b1='b1', regular case, getting objected added in line 22. It was saved under correct hash bucket and hashset could find it with equals function

		Database.TryGetValue(new FileEntry("a1"), out a1);
		Console.WriteLine($"a1='{a1}'");  //Expected output: "a1=''" because hashcode of a1 is suggesting to use hash bucket in which 'equals' function can't find the object - in result object is not accesible by get function, but not removed
	}
	public class FileEntry
	{
		public string Path { get; set; }

		public FileEntry(string path)
		{
			this.Path = path;
		}
		public override string ToString() => Path;
		public override int GetHashCode() => StringComparer.OrdinalIgnoreCase.GetHashCode(Path);
		public override bool Equals(object obj) => obj is FileEntry entry && Path.Equals(entry.Path, StringComparison.OrdinalIgnoreCase);
	}
}
```

As you can see, as long path is part of the hashcode and we are using HashSet not much can be done to avoid re-adding item after such change.